### PR TITLE
[WIP] Checksum compute

### LINKED ISSUE #45: compute sha256 during upload of objects
making the hash function configurable ideally

### PR DESCRIPTION
**Pull request by leafty**
Closes #45

### CODE REVIEW

File: app/controllers/storageBackends/ObjectBackend.scala [hunk 1]
```diff
@@ -18,16 +18,63 @@
...
 /**
  * Created by jeberle on 07.07.17.
  */
 trait ObjectBackend extends StorageBackend {
   def read( request: RequestHeader, bucket: String, name: String ): Option[Source[ByteString, _]]
-  def write( request: RequestHeader, bucket: String, name: String ): Accumulator[ByteString, Result]
+  def write( request: RequestHeader, bucket: String, name: String, hash: Option[String] ): Accumulator[ByteString, Result]
   def duplicateFile( request: RequestHeader, fromBucket: String, fromName: String, toBucket: String, toName: String ): Boolean
+
+  def processChecksum( ref: Option[String] ) = ( _: Any, checksum: Future[String] ) => checksum.map( s => println( s + " - " + ref.getOrElse( "no-hash" ) ) )
```
Comment (leafty) [inline app/controllers/storageBackends/ObjectBackend.scala]: The checksum is only used for print?

File: app/controllers/storageBackends/ObjectBackend.scala [hunk 2]
```diff
@@ -18,16 +18,63 @@
...
+        var digest: MessageDigest = MessageDigest.getInstance( "SHA-256" )
+
+        // This requests one element at the Sink startup.
+        override def preStart(): Unit = pull( in )
+
+        setHandler( in, new InHandler {
+          override def onPush(): Unit = {
+            val ii = grab( in )
+            digest.update( ii.asByteBuffer )
+            println( ii )
```
Comment (leafty) [inline app/controllers/storageBackends/ObjectBackend.scala]: Remove print statement

**Comment by leafty** [comment]
Note: I also did an `optimize import` on all controllers

**Comment by ebiiii** [comment]
closed and cherry-picked relevant commits to #68

**Pull request closed without merging**